src/main/java/org/tugraz/sysds/runtime/lineage/RewriteCPlans.java

```diff
@@ -0,0 +1,229 @@
+		pb.setInstructions(newInst);
+		LineageCacheConfig.shutdownReuse();
+		pb.execute(lrwec);
+		LineageCacheConfig.restartReuse();
+		
+		ec.setVariable(((ComputationCPInstruction)curr).output.getName(), lrwec.getVariable(LR_VAR));
+		// add this to cache
+		LineageCache.put(curr, ec);
+		}
+		catch (Exception e) {
+			throw new DMLRuntimeException("Error evaluating instruction: " + curr.toString() , e);
+		}
+		return true;
+	}
+
+	static ArrayList<Instruction> rewrite1(Instruction curr, ExecutionContext ec, ExecutionContext lrwec, MatrixBlock lastResult) 
+	{
+		// Create a transient read op over the last tsmm result
+		MetaData md = new MetaData(lastResult.getDataCharacteristics());
+		MatrixObject newmo = new MatrixObject(ValueType.FP64, "lastResult", md);
+		newmo.acquireModify(lastResult);
+		newmo.release();
+		lrwec.setVariable("lastResult", newmo);
+		DataOp lastRes = HopRewriteUtils.createTransientRead("lastResult", lastResult);
+		// Create rightIndex op to find the last matrix and the appended column
+		// TODO: For now assumption is that a single column is being appended in a loop
+		//       Need to go down the lineage to find number of columns are being appended.
+		MatrixObject mo = ec.getMatrixObject(((ComputationCPInstruction)curr).input1);
+		lrwec.setVariable("oldMatrix", mo);
+		DataOp newMatrix = HopRewriteUtils.createTransientRead("oldMatrix", mo);
+		IndexingOp oldMatrix = HopRewriteUtils.createIndexingOp(newMatrix, new LiteralOp(1), 
+				new LiteralOp(mo.getNumRows()), new LiteralOp(1), new LiteralOp(mo.getNumColumns()-1));
+		IndexingOp lastCol = HopRewriteUtils.createIndexingOp(newMatrix, new LiteralOp(1), 
+				new LiteralOp(mo.getNumRows()), new LiteralOp(mo.getNumColumns()), 
+				new LiteralOp(mo.getNumColumns()));
+		// cell topRight = t(oldMatrix) %*% lastCol
+		ReorgOp tOldM = HopRewriteUtils.createTranspose(oldMatrix);
+		AggBinaryOp topRight = HopRewriteUtils.createMatrixMultiply(tOldM, lastCol);
+		// cell bottomLeft = t(lastCol) %*% oldMatrix
+		ReorgOp tLastCol = HopRewriteUtils.createTranspose(lastCol);
+		AggBinaryOp bottomLeft = HopRewriteUtils.createMatrixMultiply(tLastCol, oldMatrix);
+		// bottomRight = t(lastCol) %*% lastCol
+		AggBinaryOp bottomRight = HopRewriteUtils.createMatrixMultiply(tLastCol, lastCol);
+		// rowOne = cbind(lastRes, topRight)
+		BinaryOp rowOne = HopRewriteUtils.createBinary(lastRes, topRight, OpOp2.CBIND);
+		// rowTwo = cbind(bottomLeft, bottomRight)
+		BinaryOp rowTwo = HopRewriteUtils.createBinary(bottomLeft, bottomRight, OpOp2.CBIND);
+		// rbind(rowOne, rowTwo)
+		BinaryOp lrwHop= HopRewriteUtils.createBinary(rowOne, rowTwo, OpOp2.RBIND);
+		DataOp lrwWrite = HopRewriteUtils.createTransientWrite(LR_VAR, lrwHop);
+		
+		// generate runtime instructions
+		ArrayList<Instruction> newInst = null;
+		if (DMLScript.EXPLAIN == ExplainType.RECOMPILE_HOPS || DMLScript.EXPLAIN == ExplainType.RECOMPILE_RUNTIME) 
+			System.out.println("LINEAGE REWRITE 1 APPLIED");
```

Comment:
Use logging instead of stdout.

Reply:
The purpose of this print is to isolate the recompile_x outputs for this rewrite from others. Besides, this prints only if explain is set to recompile_x.

Reply:
please see `RewriteAlgebraicSimplificationDynamic` - proper logging allows to set the log level from outside without recompiling. Otherwise you would overwhelm users with output. Hence, our general policy is to only send script level prints to stdout